### PR TITLE
feat: Add hepdata-lib

### PR DESCRIPTION
**Pull request by matthewfeickert**
* c.f. https://github.com/HEPData/hepdata_lib
* Recipe generated with grayskull via 'grayskull pypi hepdata-lib' and then edited to add additional information.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [N/A] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

Resolves https://github.com/HEPData/hepdata_lib/issues/117

### CODE REVIEW

**Comment by matthewfeickert** [comment]
@clelange Can you comment below if you are willing to be a maintainer on this conda-forge recipe (required before a conda-forge maintainer can review this). (c.f. https://github.com/conda-forge/staged-recipes/pull/26293#issuecomment-2101719151) As the package author I want to include you, though I am willing to help do the maintenance of hitting "merge" for future releases.

File: recipes/hepdata-lib/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/hepdata_lib-{{ version }}.tar.gz
+  sha256: 88325da7e9ee9408eafebb518117ff3c39f438048766f1a19f59ef22f6c6d1d4
+
+build:
+  # Will fail on Windows as imagemagick does not support Windows builds, but
+  # failing on a noarch is preferred by the conda-forge maintainers
+  noarch: python
```
Comment (matthewfeickert) [inline recipes/hepdata-lib/meta.yaml:15]: The Windows build will fail, but that is expected, and so leaving this as is given https://github.com/conda-forge/staged-recipes/pull/25702#discussion_r1521569904.

**Comment by clelange** [comment]
Happy to be a maintainer.

**Comment by matthewfeickert** [comment]
@conda-forge/help-python, ready for review

**Review by xhochy — APPROVED** [review]

**Pull request merged**